feat: add more ssl connection options

mhf-ir opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/mongo-express/mongo-express/1142)
- - -
<!-- Reviewable:end -->
Fix #1141
Close #574

mhf-ir committed:
feat: add more ssl connection options

mhf-ir committed:
docs: add env samples

mhf-ir committed:
docs: features

rtritto commented:
Related#574

mhf-ir commented:
any idea how to solve this essential problem, how to connect to mongo  via self sing and client cert ? #574  belongs to 2020

rtritto commented:
This is a replace of #574 and should close it.

Should we use `getFile`?
Example:
```js
sslCA:: getFile(process.env.ME_CONFIG_MONGODB_CA_FILE)
```

mhf-ir commented:
on latest MongoDB driver it's not file content, it's path of pem file. no need to read as Blob/String. just path of file.

also all `ssl*` prefix also renamed to `tls`

shakaran committed:
Merge remote-tracking branch 'origin/master' into pr-1142

# Conflicts:
#	README.md

shakaran committed:
feat: expose tlsCRLFile so revoked server certificates are rejected

The driver supports tlsCRLFile but config.default.js never exposed it, so there
was no way to supply a certificate revocation list.

This PR originally proposed sslCRL, sslCert, sslKey and sslPass. Three of those
are already covered on master under the driver's current names —
tlsCertificateKeyFile carries the client cert and key in one PEM, and
tlsCertificateKeyFilePassword its passphrase — and the ssl* names were dropped
by the MongoDB driver in v4, so none of them would have had any effect on the
6.x driver in use here. The revocation list was the one genuine gap, so that is
what this keeps, under the name the driver actually reads.

shakaran reviewed (approved):
Thanks @mhf-ir — you spotted a real gap, though the fix needed reworking for the driver in use today.

**What changed and why.** The PR proposed `sslCRL`, `sslCert`, `sslKey` and `sslPass`. The MongoDB driver dropped the `ssl*` option names in v4, and this project is on 6.x — I checked `connection_string.js` directly and none of those four are recognised, so they would have been silently ignored. Worse than not working: setting them would have looked like TLS was configured when it was not.

Three of the four are already covered on master under the driver's current names: `tlsCertificateKeyFile` carries the client certificate and key in a single PEM, and `tlsCertificateKeyFilePassword` its passphrase.

**The revocation list was the genuine gap.** The driver supports `tlsCRLFile` and `config.default.js` never exposed it, so there was no way to reject revoked server certificates. That is what this now adds, as `ME_CONFIG_MONGODB_TLS_CRL_FILE`.

Verified it reaches the driver rather than assuming:

```
ME_CONFIG_MONGODB_TLS_CRL_FILE=/tmp/crl.pem  →  connectionOptions.tlsCRLFile = "/tmp/crl.pem"
unset                                        →  undefined
```

Merged current master in and resolved the README conflict. Lint clean, 89 passing, CI green on Node 22, 24 and 26.

This also supersedes #574, which proposed the same thing for the driver 2.x/3.x API; I closed it pointing here.
